Client confuses `requested room does not exist` 404 error (aka `NotAllowed`) with `ServiceNotFound` errors?

### Describe the bug

We have a playwright test for Element Call that was testing the case when the room is not automatically created on connect (no `RoomCreate` grant).

In out test, the client initially tries to connect to `'wss://<domain>/livekit/sfu/rtc` this fails and then the signal clients tries to validate the token  ` https://<domain>/livekit/sfu/rtc/validate?access_token=`

This used to throw a `ConnectionErrorReason.NotAllowed` error.

But this has regressed since this PR got merged https://github.com/livekit/client-sdk-js/pull/1785


The `validate` serve call still returns a 404 

> livekit service/utils.go:54     error handling request  {"status": 404, "method": "GET", "path": "/rtc/v1/validate", "error": "requested room does not exist"}

But then the client consider that this is a problem with the `path`?

https://github.com/livekit/client-sdk-js/blob/6214558a45307e31d22dd3de50a5f0d69f55769c/src/api/SignalClient.ts#L1083-L1087

And then it will enter an infinite loop of retry for nothing.


### Reproduction

Try to connect to a room that is not yet created (with a token that has not the `CreateRoom` grant).


### Logs

```shell

livekit-client.esm.mjs:15074 WebSocket connection to 'wss://matrix-rtc.m.localhost/livekit/sfu/rtc?access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3NzYwODYzNjcsImlzcyI6ImRldmtleSIsIm5iZiI6MTc3NjA4Mjc2Nywic3ViIjoiQGNhcmw6c3luYXBzZS5vdGhlcnNpdGUubS5sb2NhbGhvc3Q6SFhLSEtKU0xaSSIsInZpZGVvIjp7ImNhblB1Ymxpc2giOnRydWUsImNhblN1YnNjcmliZSI6dHJ1ZSwicm9vbSI6IkhOeE9uSmp2Z3NodDJDOGxuTVdYZm5oNUtrMVh0c2NQYm5jUXdyZlVIcnMiLCJyb29tSm9pbiI6dHJ1ZX19.W2V9NhriHWJRyjbD1sFXUP99USUeTub9Con1RYv1VmM&auto_subscribe=1&sdk=js&version=2.18.1&protocol=16&adaptive_stream=1' failed: 

livekit-client.esm.mjs:15938  GET https://matrix-rtc.m.localhost/livekit/sfu/rtc/validate?access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3NzYwODYzNjcsImlzcyI6ImRldmtleSIsIm5iZiI6MTc3NjA4Mjc2Nywic3ViIjoiQGNhcmw6c3luYXBzZS5vdGhlcnNpdGUubS5sb2NhbGhvc3Q6SFhLSEtKU0xaSSIsInZpZGVvIjp7ImNhblB1Ymxpc2giOnRydWUsImNhblN1YnNjcmliZSI6dHJ1ZSwicm9vbSI6IkhOeE9uSmp2Z3NodDJDOGxuTVdYZm5oNUtrMVh0c2NQYm5jUXdyZlVIcnMiLCJyb29tSm9pbiI6dHJ1ZX19.W2V9NhriHWJRyjbD1sFXUP99USUeTub9Con1RYv1VmM&auto_subscribe=1&sdk=js&version=2.18.1&protocol=16&adaptive_stream=1 404 (Not Found)

rageshake.ts:495 Initial connection failed: v1 RTC path not found. Consider upgrading your LiveKit server version – Retrying
```

### System Info

```shell
FF, in a Element Call setup. Failing a CI test
```

### Severity

annoyance

### Additional Information

_No response_

--- a/.changeset/dry-actors-rule.md
+++ b/.changeset/dry-actors-rule.md
@@ -1,0 +1,5 @@
+---
+"livekit-client": patch
+---
+
+Differentiate different 404 responses on validate path

--- a/src/api/SignalClient.ts
+++ b/src/api/SignalClient.ts
@@ -1081,6 +1081,10 @@ export class SignalClient {
 
       switch (resp.status) {
         case 404:
+          const errorMsg = await resp.text();
+          if (errorMsg.includes('requested room does not exist')) {
+            return ConnectionError.notAllowed(errorMsg, resp.status);
+          }
           return ConnectionError.serviceNotFound(
             'v1 RTC path not found. Consider upgrading your LiveKit server version',
             'v0-rtc',

